### DISCUSSION TITLE
Both sides of ternary are rendered

### DISCUSSION THREAD

**Discussion by MikeBellika** [vercel/next.js#34982] [Help]
### What version of Next.js are you using?

12.0.4

### What version of Node.js are you using?

17.0.1

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

next start and Vercel

### Describe the Bug

In the example below the time from the server and client are compared. If they are equal, it should render a red div with the text RED. Otherwise it should render a blue div with the text BLUE.

However in practice it renders a red div, with the text BLUE. So it seems like the contents of the div are rerendered, but the div itself is not.

### Expected Behavior

Both the div and children of the div should update

### To Reproduce

Name this file example.tsx in the pages folder. Build with `npx next build` and run with `npx next start`
```typescript
import { GetStaticProps } from 'next'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default function Example(props: any) {
  if (!props || !props.serverSeconds) {
    return <></>
  }
  const clientSecond = new Date().getSeconds()
  const serverSecond = props.serverSeconds
  console.log('clientSecond', clientSecond)
  console.log('serverSecond', serverSecond)
  return clientSecond == serverSecond ? (
    <div style={{ backgroundColor: 'red', padding: '1rem' }}>RED</div>
  ) : (
    <div style={{ backgroundColor: 'blue', padding: '1rem' }}>BLUE</div>
  )
}

export const getStaticProps: GetStaticProps = async (context) => {
  const date = new Date()
  return {
    props: {
      serverSeconds: date.getSeconds(),
    },
    revalidate: 5,
  }
}
```

**Comment by ijjk** [comment]
Hi, this looks like it's due to creating a hydration mis-match. The first render on the client needs to match the initial render on the server otherwise react will fail to hydrate the tree and it can cause unexpected results in the DOM.

**Reply by icyJoseph** [reply]
Right, this is similar to https://github.com/vercel/next.js/discussions/21999 - this type of error is a pitfall of SSR. 

To do this I'd do, with strict mode off:

```tsx
import { GetStaticProps } from 'next'
import {useRef, useState, useEffect} from 'react'

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export default function Example(props: any) {
 const clientSecondRef = useRef(new Date().getSeconds());
 const clientSecond = clientSecondRef.current;
 const [done, setDone ] = useState(false);

 useEffect(() => {
   // wait until the rendering is done
   setDone(true)
 },[])

  if (!props || !props.serverSeconds) {
    return <></>
  }

 if(!done) return <></>

  const serverSecond = props.serverSeconds

  return clientSecond === serverSecond ? (
    <div style={{ backgroundColor: 'red', padding: '1rem' }}>RED</div>
  ) : (
    <div style={{ backgroundColor: 'blue', padding: '1rem' }}>BLUE</div>
  )
}

export const getStaticProps: GetStaticProps = async (context) => {
  const date = new Date()
  return {
    props: {
      serverSeconds: date.getSeconds(),
    },
    revalidate: 5,
  }
}
```
